### PR TITLE
Use default options unless it is an existing event

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
@blakeembrey, I'll give you a chance to veto this if you want.  The problem was that the default options were not being passed when we used the new event constructors.  Also, an existing event already has its options set up, so we should not be changing them in my opinion.

File: simulate-event.js [hunk 1]
```diff
@@ -305,6 +305,14 @@ exports.generate = function (type, options) {
   // `Object.defineProperty`.
   var overrides = getOverrides(eventType, options)
 
+  // Extend a new object with the default and passed in options.
+  if (!(options instanceof window.UIEvent)) {
```
Comment (sccolbert) [inline simulate-event.js]: why `UIEvent` instead of `Event`?

Comment (blink1073) [inline simulate-event.js]: This library only deals with `UIEvent` as a base class.

Comment (sccolbert) [inline simulate-event.js]: not all UI events bubble.

**Comment by blink1073** [comment]
Merging, as this is a strict improvement for now.

**Comment by blink1073** [comment]
Released as `v1.1.0`.

**Comment by blakeembrey** [comment]
LGTM. Is there any better way of structuring this perhaps? Like a function that does "without defaults" and you just forward the current function onto that. Technically this was a breaking change as a result.

**Comment by blink1073** [comment]
We could have a separate `clone()` function for existing events.

**Comment by blakeembrey** [comment]
That sounds awesome. Just thinking about the API usage, I usually prefer to avoid overloading potentially surprising behaviour - it's pretty simple to document and enable both ways.

**Comment by blink1073** [comment]
@blakeembrey, I thought about this some more over the weekend.  I think the current behavior is the right one.  The `defaults` only apply when we do not provide an option for that value, but an existing event already has all of its options set.  However, if we use `extend` on an existing object, not all of the properties are carried over because they do not pass the `hasOwnProperty` test.

**Pull request merged**